### PR TITLE
iBFT threading refactoring

### PR DESCRIPTION
**Pull request by alonmuroch**
iBFT uses a bunch of threads with on real thread management.
We've refactored the iBFT instance to be single threaded, working with an event loop pulling events from a queue

### CODE REVIEW

File: ibft/instance.go [hunk 1]
```diff
@@ -117,6 +121,9 @@ func NewInstance(opts InstanceOptions) *Instance {
 // 		broadcast ⟨PRE-PREPARE, λi, ri, inputV aluei⟩ message
 // 		set timer to running and expire after t(ri)
 func (i *Instance) Start(inputValue []byte) {
+	if !i.initialized {
+		i.Logger.Fatal("can't start instance a non  fgs instance")
```
Comment (nivBlox) [inline ibft/instance.go:125]: what the reason we proceed and not returning in this case?

File: ibft/event_loop.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// each message type has it's own pipeline of checks and actions, called by the networker implementation.
+// Internal chan monitor if the instance reached decision or if a round change is required.
+func (i *Instance) StartMessagePipeline() {
+	for {
+		if i.Stopped() {
+			return
+		}
+
+		// TODO - refactor
+		if i.MsgQueue.MsgCount(msgqueue.IBFTRoundIndexKey(i.State.Lambda, i.State.SeqNumber, i.State.Round)) > 0 {
```
Comment (nivBlox) [inline ibft/event_loop.go:53]: Need to change the of the key index name to "IBFTMessageIndexKey" or so

**Review by nivBlox — CHANGES_REQUESTED** [review]

File: ibft/instance.go (hunk 1 above)
Comment (lior-blox) [reply ibft/instance.go:125]: @alonmuroch @nivBlox let's reconsider logger type here
Fatal will crash the node. I don't think faulty of single validator/ibft should affect others

Comment (alonmuroch) [reply ibft/instance.go:125]: @lior-blox @nivBlox we will return error instead

File: ibft/event_loop.go (hunk 2 above)
Comment (alonmuroch) [reply ibft/event_loop.go:53]: @nivBlox ok

**Review by nivBlox — APPROVED** [review]

**Review by lior-blox — APPROVED** [review]

**Review by amir-blox — APPROVED** [review]

**Pull request merged**